layout of two pages

AlexeyMoseev opened the pull request.

AlexeyMoseev committed:
layout of two pages

Roman-Maksimov commented on detailed.html:
```diff
@@ -0,0 +1,131 @@
+		<link
+			href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap"
+			rel="stylesheet"
+		/>
```
В данном случае не принципиальной разницы нет, но рекомендую всегда помещать вашу таблицу стилей последней. Так у вас будет возможность переопределять third-party стили.

Roman-Maksimov commented on styles.css:
```diff
@@ -0,0 +1,407 @@
+}
+
+.search-field:focus-within {
+	border-bottom: 2px solid #e97f03;
```
Вы наверняка заметили, что при фокусе на поле, происходит небольшой скачок, из-за того, что раньше границы не было, а потом она появляется толщиной в 2px. По мимо визуального дефекта это еще и приводит к так называемому reflow (пересчету остальных элементов, расположенных ниже этого инпута).
Классически такую задачу решают следующим образом:
```css
.search-field {
  ...
  border-bottom: 2px solid transparent;
  transition: border-color 100ms ease-out;
}

.search-field:focus-within {
  border-color: #e97f03;
}
```

Roman-Maksimov commented on styles.css:
```diff
@@ -0,0 +1,407 @@
+	text-align: center;
+}
+
+.search-field:focus-within {
```
focus-within - неплохое решение, но оно не работает в IE. Переделывать не надо, просто обратите внимание. Альтернативно, можно было использовать `next-sibling`

Roman-Maksimov commented on styles.css:
```diff
@@ -0,0 +1,407 @@
+	flex-direction: column;
+	height: 272px;
+	flex: 0 0 220px;
+	margin: 0px 8px 16px 8px;
```
можно просто `margin: 0px 8px 16px;`

Roman-Maksimov commented on styles.css:
```diff
@@ -0,0 +1,407 @@
+.image {
+	width: 59px;
+	height: 79px;
+	margin-bottom: 38px;
```
Когда будете натягивать верстку на данные, скорее всего вы не будете знать точный размер картинки. Тут требуется что-то более относительное

Roman-Maksimov commented on styles.css:
```diff
@@ -0,0 +1,407 @@
+}
+
+.add-to-cart:hover {
+	background: #e95f03;
```
тень забыли

AlexeyMoseev committed:
Были исправлены следующие баги:
1)Таблицу стилей поместил последней в списке
2)Исправлил визуальный дефект поиска, который проявлялся при фокусе на элементе
3)У фотографий товара в каталоге убрал жесткую привязку к размерам
4)Кнопке "add-to-cart" добавил тень
5)В стилях у класса "item" убрал лишний код, который никак не влиял на верстку
